### PR TITLE
Feature:add github oauth login

### PR DESCRIPTION
**Pull request by hxcGit**
Related issue: https://github.com/openyurtio/yurt-dashboard/issues/7

### CODE REVIEW

**Comment by luc99hen** [comment]
@hxcGit, thank you for this great contribution! 
Please combine the four commits into one for simplicity.

File: backend/proxy_server/auth.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+// global oauth providers' client config
+var (
+	githubConfig *oauth2.Config
+)
+
+// providers' client parameters
+const (
+	GithubClientID          = "4e5058f5e68e11b91193"
+	GithubClientSecret      = "188cbe058b6bfeeb856f4962806e72342e2da3d9"
+	GithubClientRedirectURL = "http://106.14.177.57/login"
```
Comment (luc99hen) [inline backend/proxy_server/auth.go:17]: I have got a question here🤔. 
Where is the URL from? Is it a github server or selfhosted server? Does this need to be updated regularly?

Comment (hxcGit) [reply backend/proxy_server/auth.go:17]: It's a self-hosted server and here is my server IP.  You need to set this in your github account, then you will get `clientID`,`secret`. Maybe you can find it at `https://github.com/settings/developers`. For more details: `https://docs.github.com/cn/developers/apps/building-oauth-apps/creating-an-oauth-app`. As well, we need to set ClientID in front-end page.

File: backend/proxy_server/github.go [hunk 2]
```diff
@@ -61,31 +62,31 @@ func githubLoginHandler(c *gin.Context) {
...
+		logger.Warn(c.ClientIP(), "get token fail", err.Error())
+		JSONErr(c, http.StatusInternalServerError, err.Error())
 		return
 	}
 
-	githubUserInfo, errMsg := getUserGithubInfo(token)
+	githubUserInfo, err := getUserGithubInfo(token)
 	if githubUserInfo == nil {
-		logger.Warn("", "get github user info fail", errMsg)
+		logger.Warn("", "get github user info fail", err.Error())
```
Comment (luc99hen) [inline backend/proxy_server/github.go:74]: `logger.Warn(c.ClientIP(), "get github user info fail", err.Error())`

File: backend/proxy_server/github.go [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+	token, err := getOauthToken(githubConfig, postCode.Code)
+	if token == nil || token.AccessToken == "" {
+		logger.Warn(c.ClientIP(), "get token fail", err.Error())
+		JSONErr(c, http.StatusInternalServerError, err.Error())
+		return
+	}
+
+	githubUserInfo, err := getUserGithubInfo(token)
+	if githubUserInfo == nil {
+		logger.Warn("", "get github user info fail", err.Error())
```
Comment (luc99hen) [inline backend/proxy_server/github.go:74]: This can also be changed as above with `ClientIP()`

File: backend/proxy_server/github.go [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+			return nil, errors.New(errMsg)
+		}
+	} else {
+		fetchUser, err = client.GetUser(adminKubeConfig, userProfile.Mobilephone)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return fetchUser, errors.New("get local user success")
```
Comment (luc99hen) [inline backend/proxy_server/github.go:166]: error is `nil` when success

File: backend/proxy_server/github.go [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+	githubUserInfo, err := getUserGithubInfo(token)
+	if githubUserInfo == nil {
+		logger.Warn("", "get github user info fail", err.Error())
+		JSONErr(c, http.StatusInternalServerError, "get github user info fail")
+		return
+	}
+
+	localUser, err := getLocalUser(githubUserInfo)
+	if localUser == nil {
+		logger.Warn("", "get local user info fail", err.Error())
```
Comment (luc99hen) [inline backend/proxy_server/github.go:81]: Same as above

File: backend/proxy_server/github.go (hunk 3 above)
Comment (luc99hen) [reply backend/proxy_server/github.go:74]: 👌

**Pull request closed without merging**